### PR TITLE
Allow miner to implicitly close a position

### PR DESCRIPTION
**Pull request by sli-tao**
### Description
Fix bug where orders implicitly closing a position are being rejected.

ex:
long 5x -> short any lev greater than 5x

- Allow a miner to implicitly close a position by submitting an order that would take the position leverage to 0 or flip the sign of position leverage.
- Cap the order leverage for an implicit close to the same magnitude as the position net leverage.

### Related Issues (JIRA)
[Reference any related issues or tasks that this pull request addresses or closes.]

### Checklist
- [ ] I have tested my changes on testnet.
- [ ] I have updated any necessary documentation.
- [ ] I have added unit tests for my changes (if applicable).
- [ ] If there are breaking changes for validators, I have (or will) notify the community in Discord of the release.

### Reviewer Instructions
[Provide any specific instructions or areas you would like the reviewer to focus on.]

### Definition of Done
- [ ] Code has been reviewed.
- [ ] All checks and tests pass.
- [ ] Documentation is up to date.
- [ ] Approved by at least one reviewer.

### Checklist (for the reviewer)
- [ ] Code follows project conventions.
- [ ] Code is well-documented.
- [ ] Changes are necessary and align with the project's goals.
- [ ] No breaking changes introduced.

### Optional: Deploy Notes
[Any instructions or notes related to deployment, if applicable.]

/cc @mention_reviewer

### CODE REVIEW

File: vali_objects/position.py [hunk 1]
```diff
@@ -519,19 +519,24 @@ def _clamp_and_validate_leverage(self, order) -> bool:
...
-                    order.leverage *= -1
-                should_ignore_order = order.leverage == 0
-            else:
-                pass#  We are getting the leverage closer to the new boundary (decrease) so allow it
-        elif abs(proposed_leverage) < min_position_leverage:
-            if is_first_order or abs(proposed_leverage) < abs(self.net_leverage):
-                raise ValueError(f'Attempted to set position leverage below min_position_leverage {min_position_leverage}')
-            else:
-                pass  # We are trying to increase the leverage here so let it happen
+        # we only need to worry about clamping if the sign of the position leverage remains the same i.e. position does not flip and close
```
Comment (jbonilla-tao) [inline vali_objects/position.py:522]: if they trying to make a position's leverage really small, we should still block that. that can happen if the order leverage and the position leverage are different directions.

Comment (jbonilla-tao) [reply vali_objects/position.py:522]: Maybe make a explicit check for implicit close and then bypass all further checks.

Comment (jbonilla-tao) [reply vali_objects/position.py:522]: we could also just force close a position if they try taking the position leverage below the minimum position leverage

Comment (sli-tao) [reply vali_objects/position.py:522]: This change doesn't affect the case where a miner attempts to make position leverage really small e.g. long 5 -> short 4.999.

It adds an explicit check for implicit close with `self.net_leverage * proposed_leverage > 0`. That makes sure that the sign of the position leverage remains the same.

**Comment by jbonilla-tao** [comment]
Gotcha youre correct I misread

**Review by jbonilla-tao — APPROVED** [review]

**Pull request merged**